// project/events.js
var dal = require('dal/project');
var q = require('hyperquest');
var config = require('rc')('prohub');

var debounce;

exports.test = function(stream, data, token) {
  if (data.page && data.page == 'project') 
    return true;
};

exports.handler = function(stream, data, token) {

  if (data.project) {

    data.project.token = token;
    dal.getProject(token, data.project, stream, function(err, data) {

      // TODO handle error properly.
      if (err) {
        return stream.write({ 
          page: 'project', 
          error: err.message 
        });
      }
 
      data.org = config.org;
      stream.write({
        page: 'project',
        project: data
      });
    });
  }

  if (data.remove) {
    var projectId = data.remove.id;
    var repo = data.remove.repo;

    return dal.removeRepo(projectId, repo, function(err) {
      if (err) {
        return stream.write({
          page: 'project',
          error: err.message
        });
      }

      stream.write({
        page: 'project',
        remove: repo
      });
    });
  }

  if (data.repo) {
    q(data.repo.url, function(err, res) {
      if (err) {
        data.error = err;
        return stream.write(data);
      }
      dal.addRepository(data.repo, function(err) {
        if (err) {
          data.error = err.message;
          return stream.write(data);
        }
        data.status = res.headers.status;
        stream.write(data);
      });
    });
  }

};
